packages/opensrp-plans/src/helpers/dataLoaders.ts

```diff
@@ -0,0 +1,45 @@
+/** get the full product Catalogue
+ */
+
+import { store, makeAPIStateSelector } from '@opensrp/store';
+import { OpenSRPService as GenericOpenSRPService } from '@opensrp/server-service';
+import { OPENSRP_API_BASE_URL, OPENSRP_PLANS } from '../constants';
+import { fetchPlanDefinitions } from '../ducks';
+import { PlanDefinition } from '../plan-global-types';
+
+const sessionSelector = makeAPIStateSelector();
+const accessToken = sessionSelector(store.getState(), { accessToken: true });
+
+/** OpenSRP service */
+export class OpenSRPService extends GenericOpenSRPService {
+  constructor(endpoint: string, baseURL: string = OPENSRP_API_BASE_URL) {
+    super(accessToken, baseURL, endpoint);
+  }
+}
+
+/**
+ * @param {string} baseURL -  base url of api
+ * @param {OpenSRPService} service - the opensrp service
+ * @param {fetchProducts}actionCreator - Action creator; creates actions thad adds products to the store
+ *
+ * @returns {Promise<void>}
+ */
+export async function loadPlans(
```

Comment:
Why do we need to extend `GenericOpenSRPService` to add this?  Is there no generic way to do this?  Will we extend it for every single component ever?

Reply:
@moshthepitt we will be extending the service so that we can get the accesstoken in the intialization step, This will mean we will have to extend the service in every component.

The alternative to is to get the token from within the client and pass it as a prop to the components, which for me is the greater evil.

Reply:
> we will be extending the service so that we can get the accesstoken in the intialization step, This will mean we will have to extend the service in every component.

@p-netm it is not at all possible to DRY our or generalize the needed extension?

Reply:
@moshthepitt it would mean we would probably end up with a single package that has just about one extra line of code that gets and adds the token. This cannot go to the server-service, because then we would couple that with the store.

```typescript
class Service extends ExtendedService{
  constructor(..args){
    const accesstoken = '';
    super(accesstoken, ..args)
}
}
```
how do you feel about this alternative?

Reply:
Thanks for clarifying.  I think that that looks better.

Reply:
@kahummer have you seen this? ^^

Reply:
@moshthepitt Yeah I've created an issue for this :point_up_2:

Reply:
https://github.com/OpenSRP/web/issues/222

Reply:
@moshthepitt just to clarify, we do not need a package for this, its ok that we will have to duplicate that above code in all components that need to use the OpensrpService?.

Reply:
@p-netm thats not ideal, but okay with me.  However, is it consistent with he discussion here https://opensrp.slack.com/archives/CLF6Q6Q49/p1607082255023200

Reply:
yeap, seems close enough to what @kelvin-muchiri suggested,